includes/emails/class-emails.php

```diff
@@ -0,0 +1,464 @@
+			'value'    => sprintf( '<a href="mailto:%s">%s</a>', $from_email, $from_email ),
+		];
+		foreach ( $placeholders as $value ) {
+			$html = str_replace(
+				$value['template'],
+				$value['value'],
+				$html
+			);
+		}
+		$email_content_type = function() {
+			return 'text/html';
+		};
+
+		add_filter( 'wp_mail_content_type', $email_content_type );
+		$email_send_result = wp_mail( // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_mail_wp_mail
+			$to,
+			$email_config['subject'],
+			$html,
+			[
+				sprintf( 'From: %s <%s>', $email_config['from_name'], $from_email ),
+			]
+		);
+		remove_filter( 'wp_mail_content_type', $email_content_type );
+
+		if ( $switched_locale ) {
+			\restore_previous_locale();
+		}
+
+		return $email_send_result;
+	}
+
+	/**
+	 * Load a template of an email.
+	 *
+	 * @param string $type Email type.
+	 */
+	private static function load_email_template( $type ) {
+		$configs = self::get_email_configs();
+
+		if ( ! isset( $configs[ $type ], $configs[ $type ]['template'] ) ) {
+			return false;
+		}
+		return require $configs[ $type ]['template'];
+	}
+
+	/**
+	 * Does this instance support emails?
+	 */
+	public static function supports_emails() {
+		if ( ! class_exists( 'Newspack_Newsletters' ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Can email of a particular type be sent?
+	 *
+	 * @param string $type Type of the email.
+	 */
+	public static function can_send_email( $type ) {
+		if ( ! self::supports_emails() ) {
+			return false;
+		}
+		$email_config = self::get_email_config_by_type( $type );
+		if ( ! $email_config || 'publish' !== $email_config['status'] ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Get all email configs.
+	 */
+	private static function get_email_configs() {
+		return apply_filters( 'newspack_email_configs', [] );
+	}
+
+	/**
+	 * Serialize an email config.
+	 *
+	 * @param string $type Type of the email.
+	 * @param int    $post_id Email post id.
+	 */
+	private static function serialize_email( $type = null, $post_id = 0 ) {
+		if ( ! self::supports_emails() ) {
+			return false;
+		}
+
+		if ( null !== $type ) {
+			$configs = self::get_email_configs();
+			if ( ! isset( $configs[ $type ] ) ) {
+				return false;
+			}
+			$email_config = $configs[ $type ];
+		} else {
+			$email_config = [
+				'label'       => '',
+				'description' => '',
+			];
+		}
+		$html_payload = get_post_meta( $post_id, \Newspack_Newsletters::EMAIL_HTML_META, true );
+		if ( ! $html_payload || empty( $html_payload ) ) {
+			return false;
+		}
+		$serialized_email = [
+			'label'        => $email_config['label'],
+			'description'  => $email_config['description'],
+			'post_id'      => $post_id,
+			// Make the edit link relative.
+			'edit_link'    => str_replace( site_url(), '', get_edit_post_link( $post_id, '' ) ),
+			'subject'      => get_the_title( $post_id ),
+			'from_name'    => get_post_meta( $post_id, 'from_name', true ),
```

Comment:
For Reader Activation-related emails, the from name and email address should match those [set in the `Reader_Activation` class](https://github.com/Automattic/newspack-plugin/blob/master/includes/class-reader-activation.php#L1289-L1323).

Non-blocking enhancement: perhaps these from details should apply to all customizable emails, and be configurable in the UI.

Reply:
Note: https://app.asana.com/0/1202719992789620/1202857616664481/f also suggests that the emails could be easily rejected if the from email address looks suspicious to receipients' email address servers. This further highlights that the sender details should be customizable, and default to the `no-reply@sitedomain.com` address we're using for other Reader Activation emails.

Reply:
Done in 95c582a5e98d4720c164d309a78eb11e16ad25cd. The name & email can be provided in the email config, which will make them uneditable:

<img width="265" alt="image" src="https://user-images.githubusercontent.com/7383192/186637293-564e016e-3a91-4c99-9e78-e48f0c706903.png">

If these are not provided in the config, these fields just act as before, setting post meta.